src/impl/EspNowNode.cpp

```diff
@@ -159,7 +169,18 @@ bool EspNowNode::setup() {
     request.discovery_challenge = esp_random();
 
     int8_t retries = NUMBER_OF_RETRIES_FOR_DISCOVERY_REQUEST;
+
+    uint8_t channels[] = WIFI_SCAN_CHANNELS;
+    uint8_t channelCount = sizeof(channels);
+    uint8_t channelIdx = 0;
+
     while (retries-- > 0) {
+      uint8_t channel = channels[channelIdx++ % channelCount];
+      log("setting discovery channel " + std::to_string(channel), ESP_LOG_INFO);
```

Comment:
To reduce log rows, you could include this in the `log("Sending broadcast discovery request on channel...` just below

Reply:
@jamienz And I suggest doing this as well.